### ISSUE TITLE
Possibly extend TypedArray API

### ISSUE THREAD

**Issue by CryZe** [WebAssembly/memory64#109]
As per the [JS BigInt to Wasm i64 Integration](https://github.com/WebAssembly/JS-BigInt-integration), all wasm64 pointer types and equivalently sized integer types (`size_t`, `usize`, ...) are exposed to JavaScript as `BigInt`, yet the `TypedArray` family of types can not handle these values:

![image](https://github.com/user-attachments/assets/4bacf3fa-7d92-49b0-933c-a49bd03ea010)

As a workaround bindings can always downcast to `number`, but ideally this should just work.

**Comment by bvisness** [comment]
Yeah, this is unfortunate. More unfortunately, though, this will need to be taken up with TC39. (I think it wouldn't be very controversial, since I believe you can already _index_ typed arrays with BigInts.) I'll ask around within Mozilla.

**Comment by sbc100** [comment]
Side note: Apparently the indexing of the typed array with bigint is possible due to the fact the bigints will coerce to strings and element access (subscript) always take a string as the key.

**Comment by bvisness** [comment]
Well that's delightfully insane, I'd expect nothing less

**Comment by sbc100** [comment]
Closing this issue since this proposal is done and the repo is being archived. Any further discussion can take place in the spec or design repos.
